File: Spooky2Net/src/Spooky2.Core/Models/GeneratorState.cs
namespace Spooky2.Core.Models;

public enum GeneratorStatus
{
    Idle,
    Running,
    Paused,
    Held
}

public sealed record GeneratorState
{
    public required int Id { get; init; }
    public string Port { get; init; } = string.Empty;
    public GeneratorStatus Status { get; init; } = GeneratorStatus.Idle;
    public double CurrentFrequency { get; init; }
    public string CurrentProgram { get; init; } = string.Empty;
    public TimeSpan ElapsedTime { get; init; }
}

File: spooky2-huntkill-android/core/src/main/kotlin/com/spooky2/huntkill/core/model/GeneratorState.kt
package com.spooky2.huntkill.core.model

import java.time.Duration

/**
 * Runtime status of a Spooky2 generator.
 *
 * Verbatim port of the C# reference `Spooky2.Core.Models.GeneratorStatus`.
 */
enum class GeneratorStatus {
    Idle,
    Running,
    Paused,
    Held,
}

/**
 * Snapshot of a generator's current state.
 *
 * Verbatim port of the C# reference `Spooky2.Core.Models.GeneratorState`.
 * The C# `required int Id` maps to a non-defaulted [id] parameter, and
 * `TimeSpan ElapsedTime` (default `TimeSpan.Zero`) maps to [java.time.Duration].
 */
data class GeneratorState(
    val id: Int,
    val port: String = "",
    val status: GeneratorStatus = GeneratorStatus.Idle,
    val currentFrequency: Double = 0.0,
    val currentProgram: String = "",
    val elapsedTime: Duration = Duration.ZERO,
)
